src/Bridge/Doctrine/Orm/CollectionDataProvider.php

```diff
@@ -67,6 +68,9 @@ public function getCollection(string $resourceClass, string $operationName = nul
             }
         }
 
-        return $queryBuilder->getQuery()->getResult();
+        $query = $queryBuilder->getQuery();
+        //forces doctrine to not lazy load relations, see http://stackoverflow.com/questions/9848747/primary-key-of-owning-side-as-a-join-column
+        $query->setHint(Query::HINT_FORCE_PARTIAL_LOAD, true);
```

Comment:
Yes we should, but I'm not sure that a configuration parameter is the best
for this as it might be interesting to allow the configuration on the whole
"Query" object.
Anyway I'll add a parameter for now to skip lazy loading.
On Mon, 5 Sep 2016 at 12:23, Kévin Dunglas <notifications@github.com> wrote:

> In src/Bridge/Doctrine/Orm/CollectionDataProvider.php
> <https://github.com/api-platform/core/pull/717#discussion_r77501184>:
>
> > @@ -67,6 +68,9 @@ public function getCollection(string $resourceClass, string $operationName = nul
> >              }
> >          }
> >
> > -        return $queryBuilder->getQuery()->getResult();
> > +        $query = $queryBuilder->getQuery();
> > +        //forces doctrine to not lazy load relations, see http://stackoverflow.com/questions/9848747/primary-key-of-owning-side-as-a-join-column
> > +        $query->setHint(Query::HINT_FORCE_PARTIAL_LOAD, true);
>
> Maybe can we allow to change this trough a parameter the constructor?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/api-platform/core/pull/717/files/dd33381dd43ff75899b0f8a683c813b952f2ac41..179399d4fbbf3863f10d74911fdf8e921af56354#r77501184>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABQr8_aBPhbfFVnx02cqoeWeczqOVCWAks5qm-2XgaJpZM4JxbFZ>
> .
>